recipes/autoconf/all/conanfile.py

```diff
@@ -13,110 +18,77 @@ class AutoconfConan(ConanFile):
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    @property
-    def _datarootdir(self):
-        return os.path.join(self.package_folder, "bin", "share")
-
-    @property
-    def _autoconf_datarootdir(self):
-        return os.path.join(self._datarootdir, "autoconf")
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        datarootdir = self._datarootdir
-        prefix = self.package_folder
-        if self.settings.os == "Windows":
-            datarootdir = tools.unix_path(datarootdir)
-            prefix = tools.unix_path(prefix)
-        conf_args = [
-            "--datarootdir={}".format(datarootdir),
-            "--prefix={}".format(prefix),
-        ]
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder)
-        return self._autotools
-
-    def _patch_files(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-
-    @contextlib.contextmanager
-    def _build_context(self):
-        with tools.environment_append(tools.RunEnvironment(self).vars):
-            yield
+    @lru_cache(1)
+    def _autotools(self):
+        autotool = Autotools(self)
+        autotool.configure()
+        autotool.make()
+        return autotool
 
     def build(self):
-        self._patch_files()
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
+        apply_conandata_patches(self)
+        self._autotools()
 
     def package(self):
-        self.copy("COPYING*", src=self._source_subfolder, dst="licenses")
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.install()
-        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "info"))
-        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "man"))
+        autotools = self._autotools()
+        # KB-H013 we're packaging an application, place everything under bin
+        autotools.install(args=[f"DESTDIR={unix_path(self, path.join(self.package_folder, 'bin'))}"])
+        copy(self, "COPYING*", src=self.source_folder, dst=path.join(self.package_folder, "licenses"))
+        rmdir(self, path.join(self._datarootdir, "info"))
+        rmdir(self, path.join(self._datarootdir, "man"))
+
+    def _set_env(self, var_name, var_path):
+        self.output.info(f"Setting {var_name} to {var_path}")
+        self.buildenv_info.define_path(var_name, var_path)
+        setattr(self.env_info, var_name, var_path)
 
     def package_info(self):
-        self.cpp_info.libdirs = []
-        self.cpp_info.includedirs = []
```

Comment:
what happened to these 2 lines?

Reply:
The layout method should take care of that as I understand it. If I understand it wrongfully I can add them back

Reply:
layout doesn't remove libdirs & includedirs automagically AFAIK. Moreover I don't like the idea to control empty libdirs or includedirs in layout() method, it's easier for maintenance to keep all the consuming logic in package_info().

Reply:
Should I remove the layout method then?
It is my experience that this will cause issues in recipe (from source).

Or is it enough to keep the layout and revert the deletion of those two lines?